pkg/operator/staticpod/controller/installer/installer_controller.go

```diff
@@ -608,54 +604,52 @@ func setAvailableProgressingNodeInstallerFailingConditions(newStatus *operatorv1
 
 // newNodeStateForInstallInProgress returns the new NodeState
 func (c *InstallerController) newNodeStateForInstallInProgress(ctx context.Context, currNodeState *operatorv1.NodeStatus, latestRevisionAvailable int32) (status *operatorv1.NodeStatus, installerPodFailed bool, reason string, err error) {
-	ret := currNodeState.DeepCopy()
-
 	// how many previously failed of latest available revision
 	previouslyFailedLatestRevisionPods := 0
 	if currNodeState.LastFailedRevision != 0 && currNodeState.LastFailedRevision == currNodeState.TargetRevision {
 		previouslyFailedLatestRevisionPods = max(1, currNodeState.LastFailedCount)
 	}
 
+	pendingNewRevision := latestRevisionAvailable > currNodeState.TargetRevision
+
 	installerPodName := getInstallerPodName(currNodeState.TargetRevision, currNodeState.NodeName, previouslyFailedLatestRevisionPods)
 	installerPod, err := c.podsGetter.Pods(c.targetNamespace).Get(ctx, installerPodName, metav1.GetOptions{})
 	if apierrors.IsNotFound(err) {
-		// installer pod has disappeared before we saw it's termination state. Retry like if it had never existed.
-		c.eventRecorder.Warning("InstallerPodDisappeared", err.Error())
-		return currNodeState, true, fmt.Sprintf("pod %s disappeared", installerPodName), nil
-	}
-	if err != nil {
+		if !pendingNewRevision {
```

Comment:
ok, if a pod disappeared and there is a new rev just continue

Reply:
yes. It's what we always did.